cmd/viam-agent/main_windows.go

```diff
@@ -49,6 +53,11 @@ func main() {
 		return
 	}
 
+	// in service mode we have to alloc our own console to be able to send interrupts
+	if r, _, err := windows.NewLazySystemDLL("kernel32.dll").NewProc("AllocConsole").Call(); r == 0 {
```

Comment:
skimming docs for AllocConsole https://learn.microsoft.com/en-us/windows/console/allocconsole

questions:
1. 'the AllocConsole function fails if the calling process already has a console' -- guessing this is why you only do this in service mode
2. are there circumstances under which this creates a floating cmd.exe box? guessing not bc we haven't seen one
3. do you need to call freeconsole somewhere so this doesn't leak? is there a system limit on how many of these can be created?